kube/values.go

```diff
@@ -9,26 +12,62 @@ import (
 func MakeValues(roleManifest *model.RoleManifest, defaults map[string]string) (helm.Node, error) {
 	env := helm.NewEmptyMapping()
 	for name, cv := range model.MakeMapOfVariables(roleManifest) {
+		if strings.HasPrefix(name, "KUBE_SIZING_") {
+			continue
+		}
 		if !cv.Secret || cv.Generator == nil || cv.Generator.Type != model.GeneratorTypePassword {
 			ok, value := cv.Value(defaults)
 			if !ok {
 				value = "~"
 			}
-			env.AddNode(name, helm.NewScalar(value, helm.Comment(cv.Description)))
+			comment := cv.Description
+			if cv.Example != "" && cv.Example != value {
+				comment += fmt.Sprintf("\nExample: %s", cv.Example)
+			}
+			env.AddNode(name, helm.NewScalar(value, helm.Comment(comment)))
 		}
 	}
 	env.Sort()
 
-	storageClass := helm.NewEmptyMapping()
-	storageClass.AddNode("persistent", helm.NewScalar("persistent"))
-	storageClass.AddNode("shared", helm.NewScalar("shared"))
+	sizing := helm.NewEmptyMapping()
+	for _, role := range roleManifest.Roles {
+		if role.IsDevRole() || role.Run.FlightStage == model.FlightStageManual {
+			continue
+		}
+
+		entry := helm.NewEmptyMapping(helm.Comment(role.GetLongDescription()))
+
+		var comment string
+		if role.Run.Scaling.Min == role.Run.Scaling.Max {
+			comment = fmt.Sprintf("The %s role cannot be scaled.", role.Name)
+		} else {
+			comment = fmt.Sprintf("The %s role can scale between %d and %d instances.",
+				role.Name, role.Run.Scaling.Min, role.Run.Scaling.Max)
+		}
+		entry.AddInt("count", role.Run.Scaling.Min, helm.Comment(comment))
+		entry.AddInt("memory", role.Run.Memory)
+		entry.AddInt("vcpu_count", role.Run.VirtualCPUs)
+
+		diskSizes := helm.NewEmptyMapping()
+		for _, volume := range role.Run.PersistentVolumes {
+			diskSizes.AddInt(makeVarName(volume.Tag), volume.Size)
+		}
+		for _, volume := range role.Run.SharedVolumes {
+			diskSizes.AddInt(makeVarName(volume.Tag), volume.Size)
+		}
+		if len(diskSizes.Names()) > 0 {
+			entry.AddNode("disk_sizes", diskSizes.Sort())
```

Comment:
Wouldn't having the persistent + shared volumes all mixed together be a bit confusing?

Reply:
I don't have any strong opinions; I have just been following the "spec"/example from https://github.com/SUSE/cloudfoundry/blob/vladi-design/design/helm-chart-generation/scf-sizing.yml

I guess having separate lists might be nice, but (1) we don't even use shared volumes in SCF right now, and (2) we don't seem to have more than 2 volumes per role so far.

So in the interest of expediency, maybe file an enhancement request for later, so that we can get this merged in time for beta 1?

Reply:
Yeah, didn't care about it _that_ much, was just more up for discussion. I don't even hold strong enough feelings for it to want a separate bug.  Good as-is, we can revisit later if we feel like it.